app/javascript/gobierto_visualizations/webapp/containers/contracts/ContractsShow.vue

```diff
@@ -132,6 +208,32 @@ export default {
       this.contract_type = contract_type
       this.permalink = permalink
       this.assignee_routing_id = assignee_routing_id
+      this.contractor = contractor
+      this.contract_type = contract_type
+      this.start_date = start_date
+      this.end_date = end_date
+      this.award_date = award_date
+      this.batch_number = +batch_number
+      this.minor_contract = minor_contract
+      this.open_proposals_date = open_proposals_date || null
+      this.submission_date = submission_date || null
+      this.cpvs = cpvs
+      this.category_title = category_title
+    }
+
+    if (this.hasBatch) this.groupBatches()
+  },
+  methods: {
+    groupBatches() {
+      this.filterContractsBatches = this.contractsData.filter(({ title }) => title === this.title).sort((a, b) => a.batch_number - b.batch_number);
+    },
+    parseDate(value) {
+      if (!value) return
+      const convertDate = new Date(value)
+      const lang = I18n.locale || 'es';
+      const options = { year: 'numeric', month: 'short', day: 'numeric' };
+
+      return convertDate.toLocaleDateString(lang, options)
```

Comment:
You're importing above `VueFiltersMixin`, therefore you've `this.date()` fn available. Look at the implementation:

https://github.com/PopulateTools/gobierto/blob/master/app/javascript/lib/vue/filters/index.js#L17-L22

It's exactly the same you're doing, but other params. So instead of redo code, DRY. Do you need a filter? Ok, then:
```js
filters: {
  fmtdate(value) {
     return this.date(value, { year: 'numeric', month: 'short', day: 'numeric' })
  }
}
```

Use it as:
```vue
<span class="visualizations-contracts-show__text">{{ submission_date | fmtdate }}</span>
```

Reply:
Cool. It's a little tricky [calling a filter from another filter](https://forum.vuejs.org/t/calling-a-filter-from-a-filter/9551/4)
```

filters: {
    fmtdate(value) {
      return this.$options.filters.date(value, { year: 'numeric', month: 'short', day: 'numeric' })
    }
  },
  beforeCreate() {
   this.$options.filters.fmtdate = this.$options.filters.fmtdate.bind(this)
  },
```

Reply:
That's not what I said. You're mixing up things a lot. Look at the content of `VueFiltersMixin`:
```js
export const VueFiltersMixin = {
  methods: {
    translate,
    money,
    date,
    truncate,
    percent
  },
  filters: {
    translate,
    money,
    date,
    truncate,
    percent
  }
};
```

It exports both filters AND methods. What I'm saying is about using `this.date()` method INSIDE the filter function, as I pointed above

Reply:
And remove the `parseDate` method

Reply:
Done!